File: python_scripts/exon.py

class exon(object):

  def parse_exon_info(self, line):
    fields=line.split('\t')
    genomic_position=fields[0] #in the form chr?:XXX-XXX:+/-
    self.__rpkm=fields[3][:-1]
    self.__chromosome, pos_range, self.__strand=genomic_position.split(':')
    pos_range=pos_range.split('-')
    self.__start_pos=int(pos_range[0])
    self.__end_pos=int(pos_range[1])

  def __init__(self, data):
    self.parse_exon_info(data)

  def rpkm(self):
    return self.__rpkm

  def chromosome(self):
    return self.__chromosome
  
  def start_pos(self):
    return self.__start_pos

  def end_pos(self):
    return self.__end_pos

  def strand(self):
    return self.__strand

  def __str__(self):
    return str(self.__chromosome)+','+str(self.__start_pos)+','+str(self.__end_pos)

File: java/src/pojos/Exon.java
package pojos;

public class Exon {
	private String chromosome;
	private long start_pos, end_pos;
	private double rpkm;
	private String strand;
	
	public Exon(String chr, long start, long end, double rpkm, String strand){
		this.chromosome=chr;
		this.start_pos=start;
		this.end_pos=end;
		this.rpkm=rpkm;
		this.setStrand(strand);
	}
	
	public String getChromosome() {
		return chromosome;
	}

	public void setChromosome(String chromosome) {
		this.chromosome = chromosome;
	}

	public long getStart_pos() {
		return start_pos;
	}

	public void setStart_pos(long start_pos) {
		this.start_pos = start_pos;
	}

	public long getEnd_pos() {
		return end_pos;
	}

	public void setEnd_pos(long end_pos) {
		this.end_pos = end_pos;
	}

	public double getRpkm() {
		return rpkm;
	}

	public void setRpkm(double rpkm) {
		this.rpkm = rpkm;
	}

	public String getStrand() {
		return strand;
	}

	public void setStrand(String strand) {
		this.strand = strand;
	}

}
